src/Dfc.CourseDirectory.Functions/FunctionInstanceServicesCatalog.cs

```diff
@@ -1,31 +1,27 @@
-﻿using System;
-using System.Linq;
-using System.Runtime.CompilerServices;
-using Microsoft.Azure.WebJobs.Host;
-using Microsoft.Azure.WebJobs.Host.Executors;
+﻿using System.Collections.Concurrent;
 
-namespace Dfc.CourseDirectory.Functions
+public class FunctionInstanceServicesCatalog
 {
-    public class FunctionInstanceServicesCatalog : IJobActivatorEx
-    {
-        private readonly ConditionalWeakTable<IFunctionInstanceEx, IServiceProvider> _instanceServices;
-        private readonly IJobActivatorEx _innerActivator;
+    private readonly ConcurrentDictionary<Guid, IServiceProvider> _instanceServices;
 
-        public FunctionInstanceServicesCatalog(IJobActivator innerActivator)
-        {
-            _instanceServices = new ConditionalWeakTable<IFunctionInstanceEx, IServiceProvider>();
-            _innerActivator = (IJobActivatorEx)innerActivator;
-        }
+    public FunctionInstanceServicesCatalog()
+    {
+        _instanceServices = new ConcurrentDictionary<Guid, IServiceProvider>();
+    }
 
-        public IServiceProvider GetFunctionServices(Guid instanceId) =>
-            _instanceServices.SingleOrDefault(k => k.Key.Id == instanceId).Value;
+    public IServiceProvider GetFunctionServices(Guid instanceId)
+    {
+        _instanceServices.TryGetValue(instanceId, out var serviceProvider);
+        return serviceProvider;
+    }
 
-        T IJobActivatorEx.CreateInstance<T>(IFunctionInstanceEx functionInstance)
-        {
-            _instanceServices.Add(functionInstance, functionInstance.InstanceServices);
-            return _innerActivator.CreateInstance<T>(functionInstance);
-        }
+    public void AddFunctionInstance(Guid instanceId, IServiceProvider serviceProvider)
```

Comment:
What is the purpose of the addfunctioninstance and removefunction instance methods here? They didnt exist before the upgrade and dont seem to be referenced by anything 
![image](https://github.com/user-attachments/assets/4bab28a3-a68a-4ef1-a7cd-cfc812d27758)

Reply:
Yes you are right. They don't seem to be required here. I have removed these methods now